src/main/cpp/geometry_cache/mesh_store.cpp

```diff
@@ -93,6 +93,38 @@ namespace nova {
         chunk_parts_to_upload_lock.unlock();
     }
 
+    void mesh_store::remove_chunk_render_object(std::string filter_name, mc_chunk_render_object &chunk) {
+        mesh_definition def = {};
+
+        def.position = {chunk.x, chunk.y, chunk.z};
+        def.id = chunk.id;
+
+      //  chunk_parts_to_upload_lock.lock();
+        try{
+        for(auto& group : renderables_grouped_by_shader) {
```

Comment:
We can just get the renderables grouped by shader instead of iterating through them